### DISCUSSION TITLE
User-defined Operating Points

### DISCUSSION THREAD

**Discussion by Sam-Ramsahoye** [NatLabRockies/ROSCO#379] [Ideas]
Hi,

In the ROSCO toolbox, turbine steady-state operating points are implicitly derived from properties such as rated rotational speed, rated speed, cp_max etc. Moreover, properties like vs_rgn2K are set under similar assumptions.

I've previously made modifications to the toolbox to override these assumptions with bespoke operating points. Is there value in me submitting a PR with a proposed modification to Python toolbox API to allow for this?

Best,
Sam

**Comment by dzalkind** [comment]
Hi Sam,

This does sound interesting.  Can you explain how it works?  Do you prescribe pitch-TSR points to the toolbox?  Or do you provide rotor speed set points to the toolbox?  Some light documentation would be requested for future users and maintainers.

Are there any changes to the ROSCO controller library to ensure this happens in a multi-physics simulation?

There are a few features with similar capabilities on the ROSCO side, but they could use more support on the ROSCO toolbox (python) side.  So, your additions may be a nice complementary feature.  Feel free to submit a draft PR and we can go from there.

Best, Dan

**Comment by Sam-Ramsahoye** [comment]
Hi @dzalkind,

I'm actually prescribing WindSpeed-Cp and WindSpeed-TSR. I didn't need to make modifications to the ROSCO controller library (Fortran side of things).

I'll try and tidy things up and make a draft PR with some light documentation. There's a chance some of it will be redundant with recent ROSCO developments.

Best,
Sam
